exercises/error-handling/example.py

```diff
@@ -0,0 +1,22 @@
+def handle_error_by_throwing_exception():
+    raise Exception()
+
+
+def handle_error_by_returning_none(input_data):
+    try:
+        return int(input_data)
+    except ValueError:
+        return None
+
+
+def handle_error_by_returning_tuple(input_data):
+    try:
+        return (True, int(input_data))
+    except ValueError:
+        return (False, None)
+
+
+def filelike_objects_are_closed_on_exception(filelike_object):
+    with filelike_object:
+        filelike_object.close()
```

Comment:
This isn't closing the filelike object on an exception, it's closing the filelike object and then raising an exception afterwards.

A better solution might be to have a `was_open` property and `do_something` method in the `FileLike` class, with the latter just raising some form of exception. Then it should be up to the user to call `open`, `do_something`, and `close` in that order. The tests can check that an exception was raised, that `is_open` is `False`, and that `was_open` is `True`.

It'd probably be worth using the exercise's `.meta/hints.md` file to include something in the README about what methods exist in the `filelike_object`. The `.meta/hints.md` file is then included into the README when it's generated using the `configlet` - it's fine to just use copy-and-paste to put together the README, but also putting it in the hints file makes sure that it won't accidentally get lost in future.

Reply:
Also, as a learner I would expect that a context manager could be used here (eg. `with open(filelike_object) as f`:) - it's the Pythonic way to handle files. I'm not sure what would need to be implemented in the `FileLike` class to make this work though. If we can't get a context manager working, then I think that the README needs to tell learners that context managers would be the typical way to do it with files, but that they won't work with the custom class here, because otherwise there is a massive amount of room for confusion and frustration.

Reply:
The built-in `open()` method only takes a file name; you cannot implement a class to support `open(filelike_object)`. However, I like the idea of `do_something()` raising an exception. See my latest changes.